ts/src/htx.ts

```diff
@@ -4982,52 +4948,106 @@ export default class htx extends Exchange {
+         * @returns {object} an [order structure]{@link https://github.com/ccxt/ccxt/wiki/Manual#order-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const triggerPrice = this.safeNumber2 (params, 'stopPrice', 'trigger_price');
+        const stopLossTriggerPrice = this.safeNumber2 (params, 'stopLossPrice', 'sl_trigger_price');
+        const takeProfitTriggerPrice = this.safeNumber2 (params, 'takeProfitPrice', 'tp_trigger_price');
+        const isStop = triggerPrice !== undefined;
+        const isStopLossTriggerOrder = stopLossTriggerPrice !== undefined;
+        const isTakeProfitTriggerOrder = takeProfitTriggerPrice !== undefined;
         let response = undefined;
-        if (market['linear']) {
-            let marginMode = undefined;
-            [ marginMode, params ] = this.handleMarginModeAndParams ('createOrder', params);
-            marginMode = (marginMode === undefined) ? 'cross' : marginMode;
-            if (marginMode === 'isolated') {
-                if (isStop) {
-                    response = await this.contractPrivatePostLinearSwapApiV1SwapTriggerOrder (this.extend (request, params));
-                } else if (isStopLossTriggerOrder || isTakeProfitTriggerOrder) {
-                    response = await this.contractPrivatePostLinearSwapApiV1SwapTpslOrder (this.extend (request, params));
-                } else {
-                    response = await this.contractPrivatePostLinearSwapApiV1SwapOrder (this.extend (request, params));
-                }
-            } else if (marginMode === 'cross') {
-                if (isStop) {
-                    response = await this.contractPrivatePostLinearSwapApiV1SwapCrossTriggerOrder (this.extend (request, params));
-                } else if (isStopLossTriggerOrder || isTakeProfitTriggerOrder) {
-                    response = await this.contractPrivatePostLinearSwapApiV1SwapCrossTpslOrder (this.extend (request, params));
-                } else {
-                    response = await this.contractPrivatePostLinearSwapApiV1SwapCrossOrder (this.extend (request, params));
-                }
-            }
-        } else if (market['inverse']) {
-            if (market['swap']) {
-                if (isStop) {
-                    response = await this.contractPrivatePostSwapApiV1SwapTriggerOrder (this.extend (request, params));
-                } else if (isStopLossTriggerOrder || isTakeProfitTriggerOrder) {
-                    response = await this.contractPrivatePostSwapApiV1SwapTpslOrder (this.extend (request, params));
-                } else {
-                    response = await this.contractPrivatePostSwapApiV1SwapOrder (this.extend (request, params));
+        if (market['spot']) {
+            const request = await this.createSpotOrderRequest (symbol, type, side, amount, price, params);
+            response = await this.spotPrivatePostV1OrderOrdersPlace (this.extend (request, params));
```

Comment:
@Dan-krm this is the issue, we already extended the request with params inside `createOrderRequest` we should not do it again here